src/binary-encoding.adoc

```diff
@@ -0,0 +1,1600 @@
+needed to encode invocations of macro addresses in various ranges.
+
+|===
+| Address range | Bytes needed | Magnitude bits available
+
+|0 to 63
+|1
+|6
+
+|64 to 2,112
+|2
+|11
+
+|2,113 to 262,208
+|3
+|18
+
+|262,209 to 33,554,432
+|4
+|25
+|===
+
+==== Example encoding of an invocation of the macro at address `131`
+[source]
+----
+                               ┌─── The address VarUInt ends in a `1`,
+                               │    no more VarUInt bytes follow.
+                               │
+0 1 0 0 0 0 1 1  0 0 0 0 1 0 0 1
+└──┬──┘ └──┬──┘  └──────┬──────┘
+   │       │            └──────────── VarUInt containing the 7 most
+   │       └── 4 least significant    significant bits of the macro
+opcode high    bits of the macro      address
+nibble 4       address
+
+Magnitude bits: 0000100_0011
+Decoded value : 67
+Biased value  : 131
+----
+
+==== Example encoding of an invocation of the macro at address `1211`
+[source]
+----
+
+                               ┌─── The address VarUInt ends in a `1`,
+                               │    no more VarUInt bytes follow.
+                               │
+0 1 0 0 1 0 1 1  1 0 0 0 1 1 1 1
+└──┬──┘ └──┬──┘  └──────┬──────┘
+   │       │            └──────────── VarUInt containing the 7 most
+   │       └── 4 least significant    significant bits of the macro
+opcode high    bits of the macro      address
+nibble 4       address
+
+Magnitude bits: 1000111_1011
+Decoded value : 1,147
+Biased value  : 1,211
+----
+
+==== Example encoding of an invocation of the macro at address `71376`
+[source]
+----
+
+                              ┌─── The address VarUInt ends in `10`; the trailing
+                              │    zero indicates that one more VarUInt byte follows.
+                             ┌┴┐
+0 1 0 0 0 0 0 0  1 0 1 0 0 1 1 0  0 1 0 0 0 1 0 1
+└──┬──┘ └──┬──┘  └──────┬──────┘  └──────┬──────┘
+   │       │            │                └──────────── the 8 most significant bits
+   │       │            │                              of the macro address
+   │       │            │
+   │       │            └──────────── VarUInt containing the next 7 most
+   │       └── 4 least significant    significant bits of the macro
+opcode high    bits of the macro      address
+nibble 4       address
+
+Magnitude bits: 01000101_101001_0000
+Decoded value : 71,312
+Biased value  : 71,376
+----
+
+NOTE: From this point on in the document, example encodings are given in hexidecimal notation.
+
+[[booleans]]
+== Booleans
+
+`0x5E` represents boolean `true`, while `0x5F` represents boolean `false`.
+
+// XXX: Structurally, these examples should be at depth 3. However, all other example sections in the doc
+//      are at heading depth 4, which keeps them out of the TOC. These are also at depth 4 for consistency.
+
+==== Example encoding of `true`
+[source]
+----
+5e
+----
+
+==== Example encoding of `false`
+[source]
+----
+5f
```

Comment:
Super minor, I think we should standardize to upper-case for hexadecimal, we use upper elsewhere in the doc.